tika-core/src/main/java/org/apache/tika/io/CountingInputStream.java

```diff
@@ -56,7 +56,7 @@ public CountingInputStream(InputStream in) {
     @Override
     public int read(byte[] b) throws IOException {
         int found = super.read(b);
-        this.count += (found >= 0) ? found : 0;
```

Comment:
For this read(), and the following one, I wouldn't use `Math.max()`. It's really a test to see if the `super.read()` returned -1 or an actual read amount, so better would be something like `if (found != -1) { this.count += found; }`

Reply:
Good point. Just pushed a new commit. :)